### ISSUE TITLE
(TIR 14,15) Support identification of segments in input text

### ISSUE THREAD

**Issue by balmas** [alpheios-project/alignment-editor-new#80]
14. If input does not provide aligned segments, then input source and translation text is to be initially identified as a single segment of text to be aligned (no automatic segmentation)

15. If input identifies aligned segments, then this information must be preserved with the alignment.

The tokenization service is responsible for parsing of segments in input text and providing input parameters for the client to specify how segments are identified.

The alignment editor is responsible for providing  the user with an interface to provide the segmentation information as supported by the tokenizer service.

**Comment by irina060981** [comment]
> If input does not provide aligned segments, then input source and translation text is to be initially identified as a single segment of text to be aligned (no automatic segmentation)

It is done via doubleline segmentation by default (defined inside tokenization service)

>The tokenization service is responsible for parsing of segments in input text and providing input parameters for the client to specify how segments are identified.
The alignment editor is responsible for providing the user with an interface to provide the segmentation information as supported by the tokenizer service.

It was done via settings

**Comment by irina060981** [comment]
> If input identifies aligned segments, then this information must be preserved with the alignment.

@balmas , does it mean the case when we upload pre-aligned text in the format of Alpheios Alignement?
may be it is the same as this issue https://github.com/alpheios-project/alignment-editor-new/issues/76?

And I could close it and track the upper case with #76?

**Comment by balmas** [comment]
> > If input identifies aligned segments, then this information must be preserved with the alignment.
> 
> @balmas , does it mean the case when we upload pre-aligned text in the format of Alpheios Alignement?
> may be it is the same as this issue #76?
> 
> And I could close it and track the upper case with #76?

Actually, this is just the support for segments as defined by single/doubleline (plain text) or XML element (TEI) , along with the requirement than we error if the number of segments in translation and origin text don't match. So I think you can close this.

**Comment by balmas** [comment]
ready to test (although TEI testing is dependent upon #78 being finished.)

**Comment by monzug** [comment]
I did take a quick look at the TEXT details (not sure if this is part of the scope of this issue though).  in Supply the starting segment index, I left the field empty and got an error.  then I tried with 7 (words, lines or what else?) and the alignment skipped one segment where it was supposed to be.

**Comment by balmas** [comment]
> I did take a quick look at the TEXT details (not sure if this is part of the scope of this issue though). in Supply the starting segment index, I left the field empty and got an error. then I tried with 7 (words, lines or what else?) and the alignment skipped one segment where it was supposed to be.

I've made a new issue for that. Thanks!

**Comment by monzug** [comment]
so I close this one.

**Comment by monzug** [comment]
tested in safari too (singleline/doubleline for text and TEI)
